samples/utils/CommandLineUtils.cpp

```diff
@@ -162,13 +174,266 @@ namespace Utils
+        if (HasCommand("proxy_host"))
+        {
+            proxyOptions = GetProxyOptionsForMQTTConnection();
+            x509Config.ProxyOptions = proxyOptions;
+        }
+
+        provider = Aws::Crt::Auth::CredentialsProvider::CreateCredentialsProviderX509(x509Config);
+        if (!provider)
+        {
+            fprintf(stderr, "Failure to create credentials provider!\n");
+            exit(-1);
+        }
+
+        Aws::Iot::WebsocketConfig config(GetCommandRequired("signing_region"), provider);
+        clientConfigBuilder = Aws::Iot::MqttClientConnectionConfigBuilder(config);
+
+        if (HasCommand("ca_file"))
+        {
+            clientConfigBuilder.WithCertificateAuthority(GetCommand("ca_file").c_str());
+        }
+        if (HasCommand("proxy_host"))
+        {
+            clientConfigBuilder.WithHttpProxyOptions(proxyOptions);
+        }
+
+        clientConfigBuilder.WithEndpoint(GetCommandRequired("endpoint"));
+        return GetClientConnectionForMQTTConnection(client, &clientConfigBuilder);
+    }
+    std::shared_ptr<Aws::Crt::Mqtt::MqttConnection> CommandLineUtils::BuildWebsocketMQTTConnection(
+        Aws::Iot::MqttClient *client)
+    {
+        Aws::Crt::Io::TlsContext x509TlsCtx;
+        Aws::Iot::MqttClientConnectionConfigBuilder clientConfigBuilder;
+
+        std::shared_ptr<Aws::Crt::Auth::ICredentialsProvider> provider = nullptr;
+
+        Aws::Crt::Auth::CredentialsProviderChainDefaultConfig defaultConfig;
+        provider = Aws::Crt::Auth::CredentialsProvider::CreateCredentialsProviderChainDefault(defaultConfig);
+
+        if (!provider)
+        {
+            fprintf(stderr, "Failure to create credentials provider!\n");
+            exit(-1);
+        }
+
+        Aws::Iot::WebsocketConfig config(GetCommandRequired("signing_region"), provider);
+        clientConfigBuilder = Aws::Iot::MqttClientConnectionConfigBuilder(config);
+
+        if (HasCommand("ca_file"))
+        {
+            clientConfigBuilder.WithCertificateAuthority(GetCommand("ca_file").c_str());
+        }
+        if (HasCommand("proxy_host"))
+        {
+            clientConfigBuilder.WithHttpProxyOptions(GetProxyOptionsForMQTTConnection());
+        }
+
+        clientConfigBuilder.WithEndpoint(GetCommandRequired("endpoint"));
+        return GetClientConnectionForMQTTConnection(client, &clientConfigBuilder);
+    }
+    std::shared_ptr<Aws::Crt::Mqtt::MqttConnection> CommandLineUtils::BuildDirectMQTTConnection(
+        Aws::Iot::MqttClient *client)
+    {
+        Aws::Crt::String certificatePath = GetCommandRequired("cert");
+        Aws::Crt::String keyPath = GetCommandRequired("key");
+        Aws::Crt::String endpoint = GetCommandRequired("endpoint");
+
+        auto clientConfigBuilder =
+            Aws::Iot::MqttClientConnectionConfigBuilder(certificatePath.c_str(), keyPath.c_str());
+        clientConfigBuilder.WithEndpoint(endpoint);
+
+        if (HasCommand("ca_file"))
```

Comment:
I would move all of these raw strings into constants so the string itself isn't repeated all over